README.md

```diff
@@ -128,6 +128,7 @@ Asynchronous method that runs a site wide scan. Options come in the form of an o
 - `port` (number) - When the `path` is provided as a local path on disk, the `port` on which to start the temporary web server.  Defaults to a random high range order port.
 - `recurse` (boolean) - By default, all scans are shallow.  Only the top level links on the requested page will be scanned.  By setting `recurse` to `true`, the crawler will follow all links on the page, and continue scanning links **on the same domain** for as long as it can go. Results are cached, so no worries about loops.
 - `linksToSkip` (array) - An array of regular expression strings that should be skipped during the scan.
+- `filter` (function) - A function that's called for each link with the link URL as its only argument. Return `true` to check the link or `false` to skip it.
```

Comment:
I rather like the suggestion @marapper had.  Could we change the type of `linksToSkip` to support both regexes, and functions?

Reply:
Yeah that sounds good. That was my original intention, but then as a TypeScript newbie I wasn't sure how to specify that an could have multiple input types (string vs array vs function), or if that is even a good practice in the TypeScript world.

Can you give me a little guidance on how to annotate that in the schema interface?

Reply:
Is this where `any` comes in?

Reply:
Nope! You can use union types here:
https://www.typescriptlang.org/docs/handbook/advanced-types.html#union-types

So the type for the array would switch from `string []` to something like `(string | (async (link: string) => boolean))[]`